agent/pkg/models/models.go

```diff
@@ -186,3 +198,23 @@ type ExtendedCreator struct {
 	*har.Creator
 	Source *string `json:"_source"`
 }
+
+type FullEntryWithPolicy struct {
+	RulesMatched []rules.RulesMatched `json:"rulesMatched,omitempty"`
+	Entry        har.Entry            `json:"entry"`
+	Service      string               `json:"service"`
+}
+
+func (fewp *FullEntryWithPolicy) FillResultPolicy(entry MizuEntry, harEntry har.Entry) {
+	_, resultPolicyToSend := rules.MatchRequestPolicy(harEntry, entry.Service)
+	fewp.Entry = harEntry
+	fewp.RulesMatched = resultPolicyToSend
+	fewp.Service = entry.Service
+}
```

Comment:
Instead of renaming, implement the `DataUnmarshaler` interface. You just need to implement a function which has the signature defined by `DataUnmarshaler`.
I would do it like this:
```go
func (fewp *FullEntryWithPolicy) UnmarshalData(entry *MizuEntry) error {
	if err := json.Unmarshal([]byte(entry.Entry), &fewp.Entry); err != nil {
		return err
	}

	_, resultPolicyToSend := rules.MatchRequestPolicy(fewp.Entry, entry.Service)
	fewp.RulesMatched = resultPolicyToSend
	fewp.Service = entry.Service
```

Then instead of
```go
	var fullEntryWithPolicy models.FullEntryWithPolicy
	fullEntryWithPolicy.FillResultPolicy(entryData, fullEntry.Entry)
```
(in `agent/pkg/controllers/entries_controller.go`) you have
```go
	fullEntryWithPolicy := models.FullEntryWithPolicy{}
	if err := models.GetEntry(&entryData, &fullEntryWithPolicy); err != nil {
		c.JSON(http.StatusInternalServerError, map[string]interface{}{
			"error": true,
			"msg":   "Can't get entry details",
		})
	}
```

Now we have a generic way of parsing mizu entries to structs.